### PR TITLE
Invite safeties

### PR DESCRIPTION
**Pull request by ErikApption**
- added safeties for https://github.com/ssc-sp/datahub-portal/issues/535
- fixed sidebar logic

### CODE REVIEW

File: Portal/src/Datahub.Infrastructure/Services/UserEnrollmentService.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ public async Task<string> SendUserDatahubPortalInvite(string? registrationReques
         // try to see if function wrapped it in a "Value" object
         if (string.IsNullOrWhiteSpace(id))
         {
-            id = resultJson?["Value"]?["data"]?["id"]?.ToString() ?? string.Empty;
+            id = resultJson?["Value"]?["data"]?["id"]?.ToString() ?? throw new InvalidOperationException($"No ID available in response '{resultString}' from {url}");
```
Comment (yjmrobert) [inline Portal/src/Datahub.Infrastructure/Services/UserEnrollmentService.cs]: Can we wrap this in a retry?

File: Portal/src/Datahub.Portal/Layout/Sidebar.razor [hunk 2]
```diff
@@ -56,7 +56,7 @@
         _authorizedLinks = GetAuthorizedLinks(user.Id);
         
         _userHasInitiatives = !(await _userInformationService.IsUserWithoutInitiatives());
-        _isDatahubAdmin = !(await _userInformationService.IsUserDatahubAdmin());
+        _isNotDatahubAdmin = !(await _userInformationService.IsUserDatahubAdmin());
```
Comment (yjmrobert) [inline Portal/src/Datahub.Portal/Layout/Sidebar.razor]: No double negatives!

File: Portal/src/Datahub.Portal/Pages/Admin/OnboardUsers.razor [hunk 3]
```diff
@@ -205,21 +205,24 @@
...
+                    IsAdmin = false,
+                    IsDataApprover = false,
+                    User_Name = chosenUser.Mail.ToLower()
+                };
+            _chosenUsers.Add(user);
+            _inputValue = string.Empty;
+            Context.Project_Users.Add(user);
+            await _achievementService.AddOrIncrementTelemetryEvent(DatahubUserTelemetry.TelemetryEvents.UserSentInvite);
+            await RefreshData(true);
+        }
```
Comment (yjmrobert) [inline Portal/src/Datahub.Portal/Pages/Admin/OnboardUsers.razor]: Let's check at the start and throw an exception before continuing

**Review by yjmrobert — CHANGES_REQUESTED** [review]

**Review by iroelperez — APPROVED** [review]
I don't like the double negatives either. :)

**Review by yjmrobert — APPROVED** [review]

**Pull request merged**